src/test/java/games/strategy/debug/LogReaderTest.java

```diff
@@ -0,0 +1,66 @@
+package games.strategy.debug;
+
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+
+import java.io.IOException;
+import java.io.PrintStream;
+import java.lang.reflect.InvocationTargetException;
+
+import javax.swing.JTextArea;
+import javax.swing.SwingUtilities;
+
+import org.junit.experimental.extensions.MockitoExtension;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+
+@ExtendWith(MockitoExtension.class)
+public class LogReaderTest {
+
+  @Mock
+  private PrintStream stream;
+  @Mock
+  private JTextArea area;
+  @Mock
+  private GenericConsole console;
+
+  @Test
+  public void testStreamSplittingArray() throws IOException, InvocationTargetException, InterruptedException {
+    final LogReader reader = new LogReader(stream, area, Boolean.TRUE::booleanValue, console);
+    final String testString = "Some Test String";
+    final byte[] testByteArray = testString.getBytes();
+    reader.getStream().write(testByteArray);
+    verify(stream).write(testByteArray, 0, testByteArray.length);
+    SwingUtilities.invokeAndWait(() -> verify(area).append(testString));
+  }
+
+  @Test
+  public void testStreamSplittingInt() throws IOException, InvocationTargetException, InterruptedException {
+    final LogReader reader = new LogReader(stream, area, Boolean.TRUE::booleanValue, console);
+    final String testString = " ";
+    final int testInt = testString.codePointAt(0);
+    reader.getStream().write(testInt);
+    verify(stream).write(testInt);
+    SwingUtilities.invokeAndWait(() -> verify(area).append(" "));
+  }
+
+  @Test
+  public void testWriteInvisible() throws IOException, InvocationTargetException, InterruptedException {
+    final LogReader reader = new LogReader(stream, area, Boolean.FALSE::booleanValue, console);
+    final String testString = "Some Test String";
+    reader.getStream().write(testString.getBytes());
+    SwingUtilities.invokeAndWait(() -> verify(area).append(testString));
```

Comment:
One thing I forgot to comment on...  If we successfully adopt TestFX for JavaFX, we should look into a similar framework for Swing so we have a standard way to do stuff like this.

Reply:
Or we get JavaFX fully functional before that happens ^^